src/WebJobs.Extensions.DurableTask/EventSourceListener.cs

```diff
@@ -60,7 +63,20 @@ protected override void OnEventWritten(EventWrittenEventArgs eventData)
                && eventData.EventSource.Name == "DurableTask-Core"
                && eventData.Level == EventLevel.Verbose))
             {
-                this.logger.Log(eventData);
+                try
+                {
+                    this.logger.Log(eventData);
+                }
+                catch (Exception ex)
+                {
+                    this.traceHelper.ExtensionInformationalEvent(
+                    hubName: string.Empty,
+                    instanceId: string.Empty,
+                    functionName: string.Empty,
+                    message: $"The Linux Logger failed to log, logs might be lost.",
+                    writeToUserLogs: true);
+                    throw new Exception("The Linux Logger failed to log, logs might be lost", ex);
```

Comment:
In general, I don't know how helpful the new Exception wrapping this would be. You can preserve the original exception (including the stacktrace) by just having `throw;` in your `catch` block.